src/include/mpir_dataloop.h

```diff
@@ -120,11 +120,10 @@
  *
  * Note: The flag to MPIR_Segment_init() was originally simply "hetero"
  * and was a boolean value (0 meaning homogeneous). Some MPICH code
- * may still rely on HOMOGENEOUS being "0" and HETEROGENEOUS being "1".
+ * may still rely on HOMOGENEOUS being "0".
  */
 #define DLOOP_DATALOOP_HOMOGENEOUS   0
-#define DLOOP_DATALOOP_HETEROGENEOUS 1
-#define DLOOP_DATALOOP_ALL_BYTES     2
+#define DLOOP_DATALOOP_ALL_BYTES     1
```

Comment:
Why is "HOMOGENEOUS" retained?

Reply:
What should I rename that macro to? The code does make a distinction between DLOOP_DATALOOP_HOMOGENEOUS and DLOOP_DATALOOP_ALL_BYTES.

Reply:
I renamed to it to `DLOOP_DATALOOP_REGULAR` to distinguish it from `DLOOP_DATALOOP_ALL_BYTES`.

Reply:
Can you use `DEFAULT` instead of `REGULAR`. "Regular" to me says something about the nature of the data, which might mislead readers of the code.